OpenKeychain/src/main/java/org/sufficientlysecure/keychain/ui/CreateKeyEmailFragment.java

```diff
@@ -143,6 +138,68 @@ public void onClick(View v) {
         return view;
     }
 
+    /**
+     * Checks if a given email is valid
+     *
+     * @param email
+     * @param additionalEmail
+     * @return
+     */
+    private boolean checkEmail(String email, boolean additionalEmail) {
+        //check for email format or if the user did any input
+        if (!isEmailFormatValid(email)) {
+            Notify.create(getActivity(),
+                    getString(R.string.create_key_email_invalid_email),
+                    Notify.LENGTH_LONG, Notify.Style.ERROR).show();
+            return false;
+        }
+
+        //check for duplicated emails
+        if (!additionalEmail && isEmailDuplicatedInsideAdapter(email) || additionalEmail &&
+                mEmailEdit.getText().length() > 0 && email.equals(mEmailEdit.getText().toString())) {
+            Notify.create(getActivity(),
+                    getString(R.string.create_key_email_already_exists_text),
+                    Notify.LENGTH_LONG, Notify.Style.ERROR).show();
+            return false;
+        }
+
+        return true;
+    }
+
+    /**
+     * Checks the email format
+     * Uses the default Android Email Pattern
+     *
+     * @param email
+     * @return
+     */
+    private boolean isEmailFormatValid(String email) {
+        Pattern emailPattern = Patterns.EMAIL_ADDRESS;
+
+        //check for email format or if the user did any input
+        return !(email.length() == 0 || !emailPattern.matcher(email).matches());
+    }
+
+    /**
+     * Checks for duplicated emails inside the additional email adapter.
+     *
+     * @param email
+     * @return
+     */
+    private boolean isEmailDuplicatedInsideAdapter(String email) {
```

Comment:
Isn't this already checked by the recycler list? I already get an error when adding a duplicated item in the current development branch.